app/src/main/java/com/jxw/git_hub_users/view/DetailViewActivity.java

```diff
@@ -15,7 +32,116 @@ protected void onCreate(Bundle savedInstanceState) {
         Toolbar toolbar = findViewById(R.id.toolbar);
         setSupportActionBar(toolbar);
 
+
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        Intent incomingIntent = getIntent();
+
+        progressDialog = new ProgressDialog(this);
+        progressDialog.setMessage("Loading...");
+        progressDialog.show();
+
+
+
+        if (savedInstanceState != null) {
+            this.ghUserInfo = savedInstanceState.getParcelable(USER_INFO_KEY);
+            if (this.ghUserInfo != null) {
+                this.displayUserProfile(this.ghUserInfo);
+            } else {
+                fetchData(incomingIntent);
+            }
+        } else {
+            fetchData(incomingIntent);
+        }
+
+        Log.i(TAG, "ON-CREATE IS CALLED");
+
+
+    }
+
+    private void fetchData(Intent newIntent) {
+        // HANDLING INCOMING INTENT
+        if (newIntent.hasExtra("userName")) {
+            String userName = newIntent.getStringExtra("userName");
+
+            UserProfilePresenter profilePresenter = new UserProfilePresenter(this);
+            profilePresenter.getUserProfile(userName);
+        } else {
+            Toast toast = Toast.makeText(getApplicationContext(), "Invaid Username ID, Please try again", Toast.LENGTH_LONG);
+            toast.show();
+        }
+    }
+
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+
+        outState.putParcelable(USER_INFO_KEY, this.ghUserInfo);
+        Log.i(TAG, "ON-SAVE-INSTANCE-STATE IS CALLED");
+    }
+
+    @Override
+    public void handleError() {
+        progressDialog.dismiss();
+        Toast.makeText(this, "Something Went Wrong. Please, Try Again", Toast.LENGTH_SHORT).show();
+    }
+
+    @Override
+    public void displayUserProfile(UserProfile userProfile) {
```

Comment:
Kindly refactor this method to have at most 25 lines of code.

Reply:
Alright ... I will do just that

Reply:
Hi @petermwash 
I have fixed all the requested changes with this *PR*